### PR TITLE
Add all missing options to rascaline-torch calculators

### LINKED ISSUE #218: Add samples/properties/keys selection to rascaline-torch
This was left out of #200 which was large enough, but should be made available for API consistency.

### PR DESCRIPTION
**Pull request by Luthaf**
Fix #218

----
:books: Documentation preview :books:: https://rascaline--227.org.readthedocs.build/en/227/

### CODE REVIEW

File: rascaline-torch/include/rascaline/torch/calculator.hpp [hunk 1]
```diff
@@ -15,6 +15,60 @@ class RascalineAutograd;
 class CalculatorHolder;
 using TorchCalculator = torch::intrusive_ptr<CalculatorHolder>;
 
+class CalculatorOptionsHolder;
+using TorchCalculatorOptions = torch::intrusive_ptr<CalculatorOptionsHolder>;
+
+/// Options for a single calculation
+class RASCALINE_TORCH_EXPORT CalculatorOptionsHolder: public torch::CustomClassHolder {
```
Comment (agoscinski) [inline rascaline-torch/include/rascaline/torch/calculator.hpp:22]: sorry could not figure what this syntax is doing. What is this macro doing to the class declaration?

File: rascaline-torch/src/calculator.cpp [hunk 2]
```diff
@@ -288,3 +296,99 @@ metatensor_torch::TorchTensorMap rascaline_torch::register_autograd(
...
+    if (selection.isNone()) {
+        // all good
+    } else if (selection.isCustomClass()) {
+        // check if we have either a Labels or TensorMap
+        try {
+            selection.toCustomClass<metatensor_torch::LabelsHolder>();
+        } catch (const c10::Error&) {
+            if (tensormap_ok) {
+                try {
+                    selection.toCustomClass<metatensor_torch::TensorMapHolder>();
```
Comment (agoscinski) [inline rascaline-torch/src/calculator.cpp:317]: not sure what is happening here and what exactly `tensormap_ok` is expressing. What confuses me is how you can make a Labels object and a TensorMap out of a selection object. Can you add some doc?

File: rascaline-torch/include/rascaline/torch/calculator.hpp (hunk 1 above)
Comment (Luthaf) [reply rascaline-torch/include/rascaline/torch/calculator.hpp:22]: This is originally a Windows thing, where you need to explicitly declare which classes/functions should be exported & visible from a DLL/shared library. We also add a flag to mimic this behavior on Linux/macOS (`-fvisibility=hidden`).

The macro is generated by CMake (check `target/tmp/torch-tests/include/rascaline/torch/exports.h` for what it looks like on your system) to expand to the right attribute both when compiling the code & when using the code from an external project.

Comment (agoscinski) [reply rascaline-torch/include/rascaline/torch/calculator.hpp:22]: okay I kind of get this

**Pull request merged**